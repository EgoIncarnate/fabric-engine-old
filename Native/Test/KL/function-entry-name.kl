// This function is accessible from KL as "foo" but gets
// compiled to the global symbol "bar"
function Scalar foo( Scalar x ) = "bar" {
  return 1.0/x;
}

function entry() {
  report foo(2);
}
