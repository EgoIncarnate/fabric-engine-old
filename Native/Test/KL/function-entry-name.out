GlobalList
  Function( 'foo' ('bar'), Scalar:r-value )
    ParamList
      Param( 'x', Scalar, r-value )
    CompoundStatement
      StatementList
        ReturnStatement
          BinOp( DIV )
            ConstScalar( 1 )
            Var('x')
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        Report
          Call('foo')
            ArgList
              ConstSize( 2 )
0.5
